core/hydromet_stratified.py

```diff
@@ -557,4 +748,119 @@ def plot_max_potential_retention_cond_runoff(GEV_parameters_E: np.ndarray, PMP:
+    obtain a runoff curve for each recurrence interval. Also applies to events beyond published Atlas 14 values. 
+    """
+    runoff_distros1 = {}
+    prep_curves = pd.DataFrame(columns = hydro_events)
+    hyeto_graphs = np.where(df_weights_rainfall.index.to_numpy().astype(int)<1000, df_weights_rainfall.index.to_numpy().astype(int),  1000).tolist()
+    for event, hyetograph in zip(hydro_events, hyeto_graphs):
+        dist_df = get_hyeto_input_data_nrcs(nrcs_precip_table_dir, hyetograph, display_print)
+        dist_df['precip'] = dist_df['ratio']*df_weights_rainfall['P_Mean_in'].loc[event]
+        s = S_24hr(CN)
+        ia = IA_24hr(s)
+        runoff_distros1[event] = excess_precip(dist_df,ia, s)
+        prep_curves[event] = dist_df['hyeto_input']
+    return prep_curves
+    
+def precip_to_runoff_h3(hydro_events:np.ndarray ,nrcs_precip_table_dir: pl.WindowsPath,
+                     precip_data: pd.DataFrame,df_weights_rainfall: pd.DataFrame, display_print = False) -> pd.DataFrame:
+    """Takes the events, precipitation data, nrcs temporal distribution, selected CNs and applies the CN 
+    reduction method to obtain a runoff curve for each recurrence interval. 
+    """  
+    hyeto_graphs = np.where(hydro_events <1000, hydro_events, 1000).astype(int).tolist()
+    runoff_distros1 = {}
+    hydro_events = df_weights_rainfall.index.to_numpy().tolist()
+    prep_curves = pd.DataFrame(columns = hydro_events)
+    for event, hyetograph in zip(hydro_events, hyeto_graphs):
+        dist_df = get_hyeto_input_data_nrcs(nrcs_precip_table_dir, hyetograph, display_print)
+        dist_df['precip'] = dist_df['ratio']*df_weights_rainfall['Rainfall'].loc[event]
+        s = df_weights_rainfall['Avg. S'].loc[event]
+        ia = IA_24hr(s)
+        runoff_distros1[event] = excess_precip(dist_df,ia, s)
+        prep_curves[event] = dist_df['hyeto_input']
+    return prep_curves
+    
+def precip_to_runoff_h4(hydro_events:np.ndarray ,atlas14_precip_table_dir: pl.WindowsPath,
+                     precip_data: pd.DataFrame,df_weights_rainfall: pd.DataFrame, display_print = False) -> pd.DataFrame:
+    """Takes the events, precipitation data, atlas 14 temporal distribution, selected CNs and applies the 
+    CN reduction method to obtain a runoff curve for each recurrence interval. 
+    """  
+    Atlas14_hyetographs = ['q1', 'q2', 'q3', 'q4']
+    hyeto_graphs = np.where(hydro_events <1000, hydro_events, 1000).astype(int).tolist()
+    hydro_events = df_weights_rainfall.index.to_numpy().tolist()
+    column_names = []
+    
+    for event in hydro_events:
+        for hyetograph in Atlas14_hyetographs:
+            column_names.append(event+'_'+hyetograph)
+            
+    prep_curves = pd.DataFrame(columns = column_names)
+    prep_weights = pd.DataFrame(index = column_names, columns= ['Event Weight'])
+    
+    for event in hydro_events:
```

Comment:
the hyetographs are loaded into memory 96 different times, it would be more efficient to write the for-loop so that it only loads them 4 times.

Reply:
changed to zip

Reply:
No need to change it again, but its still loading the table each time. You would need to pull the get_hyeto_input_data_atlas function above the for event in hydro_events loop and set it equal to two variables that you then make copies of within the for event in hydro_events loop. The tables are small so it takes no time to load them into memory.